template/rust-http/Dockerfile

```diff
@@ -1,32 +1,43 @@
-FROM openfaas/of-watchdog:0.7.2 as watchdog
-
-FROM rust:1.37-alpine as builder
+FROM ghcr.io/openfaas/of-watchdog:0.9.11 as watchdog
+FROM rust:1.56-alpine as build
 
 WORKDIR /home/rust
 
 # Copy all the sources
-COPY function ./function
 COPY main ./main
+COPY function ./function
 
-RUN cd main && cargo build --target x86_64-unknown-linux-musl --release
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
+    cd main && cargo build --target x86_64-unknown-linux-musl --release
 
-FROM alpine:3.11 as runner
+FROM alpine:3.17.2 as ship
+
+COPY --from=watchdog /fwatchdog /usr/bin/fwatchdog
+RUN chmod +x /usr/bin/fwatchdog
 
+ARG ADDITIONAL_PKG=""
 # Install packages and add non-root user
-RUN apk --no-cache add curl ca-certificates \
+RUN apk --no-cache add curl ca-certificates libgcc gcompat ${ADDITIONAL_PKG} \
```

Comment:
Why is gcc needed?

Reply:
gcc was in fact not need needed. I had picked up the wrong binary from the build step I just realised.